core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1976,6 +1976,56 @@ private[spark] object Utils extends Logging {
     }
   }
 
+  /**
+   * Returns a qualified path object.
+   * @param defaultUri default uri corresponding to the filesystem provided.
+   * @param workingDir the working directory for the particular child path wd-relative names.
+   * @param path Path instance based on the path string specified by the user.
+   * @return Returns a qualified path object
+   */
+  def makeQualified(defaultUri: URI, workingDir: Path, path: Path): Path = {
+    val pathUri = {
+      if (!path.isAbsolute()) {
+        new Path(workingDir, path).toUri()
+      }
+      else {
+        path.toUri()
+      }
+    }
+    val newPaths = {
+      if (pathUri.getScheme == null || pathUri.getAuthority == null &&
+        defaultUri.getAuthority != null) {
+        val scheme = {
+          if (pathUri.getScheme == null) defaultUri.getScheme else pathUri.getScheme
+        }
+        val authority = {
+          if (pathUri.getAuthority == null) {
+            if (null == defaultUri.getAuthority) {
```

Comment:
Use `== null`; there's no possibility of accidentally writing `foo = null` as it does not evaluate to a boolean